pageserver/src/tenant/timeline.rs

```diff
@@ -796,6 +706,7 @@ impl Timeline {
 
         // Define partitioning schema if needed
 
+        // FIXME: the match should only cover repartitioning, not the next steps
```

Comment:
Pre-existing situation for a long time already, right? Just remove this FIXME and create a follow-up issue if you want to fix it soon.

Reply:
I don't think I will create issues for all of these. I would rather just test this out in staging and then any issues that come up, while handling these follow-ups.

Reply:
So, you're planning to fix this long-standing issue?

Reply:
I don't think this counts as a follow-up to this PR.

If you insist on treating it as such, at a minimum please include it in the 'follow-ups' list in the PR description.

Reply:
Mentioned it next to "compaction async again" in the PR desc.